live_archive.py

```diff
@@ -0,0 +1,285 @@
+
+        log_file.seek(0)
+        json.dump(message_list, log_file, indent=4)
+        log_file.truncate()
+
+
+def add_thread_reply(channel_id, thread_ts, reply_user, reply_ts):
+    """Adds a thread reply object to an archived message"""
+    with open(log_file_path(channel_id, thread_ts), 'r+') as log_file:
+        message_list = json.load(log_file)
+        message = next((m for m in message_list if m['ts'] == thread_ts), None)
+        if message is not None:
+            if "reply_count" in message:
+                message['reply_count'] += 1
+            else:
+                message['reply_count'] = 1
+
+            reply = {"user": reply_user, "ts": reply_ts}
+            if "replies" in message:
+                message['replies'].append(reply)
+            else:
+                message['replies'] = [reply]
+
+            if "reply_users" in message:
+                if reply_user not in message['reply_users']:
+                    message['reply_users'].append(reply_user)
+                    if "reply_users_count" in message:
+                        message['reply_users_count'] += 1
+                    else:
+                        message['reply_users_count'] = len(message['reply_users'])
+            else:
+                message['reply_users'] = [reply_user]
+                message['reply_users_count'] = 1
+                # TODO: Cover the weird case of no reply users but an already existing reply_users_count
+                # Though not sure how relevant this is, as slack-export-viewer does not seem to care
+        else:
+            print(f"Received reply to thread (channel={channel_id}, ts={reply_ts}) but have not found"
+                   f"original thread with ts = {thread_ts}.", file=sys.stderr)
+        # Go back to start and write
+        log_file.seek(0)
+        json.dump(message_list, log_file, indent=4)
+        log_file.truncate()
+
+
+def rename_channel(channel_id, old_name, new_name):
+    """Renames channel folders and updates channel.json"""
+    channel_list_path = archive_path / "channels.json"
+
+    with open(channel_list_path, 'r+') as channel_list:
+        old_channel_list = json.load(channel_list)
+        channel = next((ch for ch in old_channel_list if ch['id'] == channel_id), None)
+        if channel is not None:
+            channel['name'] = new_name
+        else:
+            print(f"Renamed channel (id={channel_id}) from {old_name} to {new_name},"
+                   f"but have not found channel with id on channels.json", file=sys.stderr)
+        channel_list.seek(0)
+        json.dump(old_channel_list, channel_list, indent=4)
+        channel_list.truncate()
+
+    # print(f"Channel rename event from {old_name} to {new_name}")
+    if old_name is not None:
+        old_path = archive_path / old_name
+        new_path = archive_path / new_name
+        old_path.rename(new_path)
+    else:
+        print(f"Has not found channel with id = {channel_id} in channels.json")
```

Comment:
```suggestion
        print(f"Warning: Attempted to rename channel id {channel_id}, but it doesn't exist")
```

Something like this might be a bit easier to understand